jujugui/static/gui/src/app/user/user.js

```diff
@@ -30,6 +41,23 @@ const User = class User {
     this.sessionStorage = cfg.sessionStorage || sessionStorage;
     this.localStorage = cfg.localStorage || localStorage;
     this._external = cfg.externalAuth || null;
+    this.expiration = cfg.expiration || getExpirationDate();
+  }
+
+  get expiration() {
+    return new Date(this.sessionStorage.getItem('expiration'));
```

Comment:
Might be worth putting a note here about what's going on. All the other code assumes a Date, but this doesn't, it's explicitly created.